squirrel/iterstream/base.py

```
@@ -40,14 +41,20 @@ def __iter__(self) -> t.Iterator:
     def compose(self, constructor: t.Type[Composable], *args, **kw) -> Composable:
         """Compose the items in the stream with items generated by `constructor`"""
         assert "source" not in kw
-        kw["source"] = self
-        return constructor(*args, **kw)
+        if "source" in inspect.signature(constructor).parameters:
+            assert inspect.signature(constructor).parameters["source"].default is None
```

Please raise an informative exception

---

Not sure if ValueError is the right Exception here